ueberdosis/tiptap
# Fix/insert content at block insertions

## Changes Overview
This PR fixes an issue that was brought up by a community member about how the `insertContent` and `insertContentAt` commands always split text nodes when **the selection starts at the beginning of the text**. This would always lead to the inserted content having an empty paragraph inserted above it.

In the following example I'll try to point out the issue more

```html
<p>Hello World</p>
```

Let's say your cursor right now is at the start of `Hello World` and you would insert an image, the resulting HTML previously would return

```html

 

<p></p>
<img src="https://image.domain/path.jpg" alt="My image" />
<p>Hello World</p>
```

After this fix the HTML will result to this:

```html
<img src="https://image.domain/path.jpg" alt="My image" />
<p>Hello World</p>
```

When the content is inserted between existing text, it will just work as usual via splitting the paragraph/text node apart.

## Implementation Approach
Before we run the actual inserting of block node content, I added a check to see if

1. The cursor is currently at the start of it's parent node
2. The cursor is inside a text or textblock node

If both requirements are met, we jump out of the node via `from =- 1` to prepend said node before the current text block.

## Testing Done
I added two test cases for those scenarios and both run valid + did manual tests.

## Verification Steps
1. Check the tests (should run through)
2. Clone manually if needed & open the `insertContent` demo

## Additional Notes
This is a **major** change as it could break existing insertContent usages.

## Checklist
